src/async/GenericMuxedInterruptDistributor.cc

```diff
@@ -3,27 +3,718 @@
+  : MuxedInterruptDistributor(parent), _path(registerPath.c_str()) {
+    // Set required registers
+    optionRegisterSettings.set(ISR);              // Ensure that the required option ISR is always set.
+    if(not optionRegisterSettings.test(IMaskR)) { // Ensure IMaskR or IER is on
+      optionRegisterSettings.set(IER);
+    }
+    // Note that we currently don't allowing IMaskR, but use of it gets caught later.
+
+    // Here we could explicitly note that we're ignoring IPR with optionRegisterSettings.reset(IPR)
+
+    /*----------------------------------------------------------------------------------------------------------------*/
+    steriliseOptionRegisterSettings(optionRegisterSettings, parent->getId());
+
+    /*----------------------------------------------------------------------------------------------------------------*/
+    _isr = _backend->getRegisterAccessor<uint32_t>(_path / getOptionRegisterStr(ISR), 1, 0, {});
+
+    _ierIsReallyImaskr = optionRegisterSettings.test(IMaskR);
+    _ier = _backend->getRegisterAccessor<uint32_t>(
+        _path / getOptionRegisterStr(_ierIsReallyImaskr ? IMaskR : IER), 1, 0, {});
+
+    // Set the clear/acknowledge register {ICR, IAR, ISR}
+    if(optionRegisterSettings.test(ICR)) { // Use ICR as the clear register
+      _icr = _backend->getRegisterAccessor<uint32_t>(_path / getOptionRegisterStr(ICR), 1, 0, {});
+    }
+    else if(optionRegisterSettings.test(IAR)) { // Use IAR as the clear register
+      _icr = _backend->getRegisterAccessor<uint32_t>(_path / getOptionRegisterStr(IAR), 1, 0, {});
+    }
+    else { // Use ISR to clear interrupts using its write 1 to clear feature
+      _icr = _backend->getRegisterAccessor<uint32_t>(_path / getOptionRegisterStr(ISR), 1, 0, {});
+    }
+
+    _hasMer = optionRegisterSettings.test(MER) or optionRegisterSettings.test(MIE) or optionRegisterSettings.test(GIE);
+    if(_hasMer) {
+      GmidOptionCode _optionMerMieGie =
+          optionRegisterSettings.test(MIE) ? MIE : (optionRegisterSettings.test(GIE) ? GIE : MER);
+      _mer = _backend->getRegisterAccessor<uint32_t>(_path / getOptionRegisterStr(_optionMerMieGie), 1, 0, {});
+    }
+
+    // steriliseOptionRegisterSettings ensures that either SIE and CIE are both enabled or neither are
+    _haveSieAndCie = optionRegisterSettings.test(SIE);
+    if(_haveSieAndCie) {
+      _sie = _backend->getRegisterAccessor<uint32_t>(_path / getOptionRegisterStr(SIE), 1, 0, {});
+      _cie = _backend->getRegisterAccessor<uint32_t>(_path / getOptionRegisterStr(CIE), 1, 0, {});
+    }
+
+    /*----------------------------------------------------------------------------------------------------------------*/
+    // Check register readability
+    if(!_isr->isReadable()) {
+      throw ChimeraTK::runtime_error("DummyMuxedInterruptDistributor: Handshake register not readable: " +
+          explainOptCode(getOptionRegisterEnum(_isr->getName())));
```

Comment:
I think the DummyMuxedInterruptDistributor is a left-over

Reply:
fixed